### PR TITLE
Fix TinyMCE edit link when link contains html

### PR DESCRIPTION
**Pull request by satrun77**
## Description

If you have a link in tinymce like `<a href=""><span>Title</span></a>`, then you can edit the link. The HTML inside the link tag can be any tag. 

The issue is that the selected node is the inside HTML tag instead of the link tag.

I'm not 100% keen on the solution using `while` but this is legacy code, so might be ok! Alternative code is welcomed :) 

The PR https://github.com/silverstripe/silverstripe-cms/pull/2993 depends on this PR.

## Manual testing steps

1. Install fresh CMS
2. Edit a page
3. Add a link (internal or external)
4. Edit HTML code in TinyMCE, add any HTML tags inside the link
5. Close the HTML code view in TInyMCE
6. Try to edit the link, it won't show the popup form.

## Issues

- https://github.com/silverstripe/silverstripe-cms/issues/2439

## Pull request checklist

- [x] The target branch is correct
    - See [picking the right version](https://docs.silverstripe.org/en/contributing/code/#picking-the-right-version)
- [x] All commits are relevant to the purpose of the PR (e.g. no debug statements, unrelated refactoring, or arbitrary linting)
    - Small amounts of additional linting are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
- [x] The commit messages follow our [commit message guidelines](https://docs.silverstripe.org/en/contributing/code/#commit-messages)
- [x] The PR follows our [contribution guidelines](https://docs.silverstripe.org/en/contributing/code/)
- [x] Code changes follow our [coding conventions](https://docs.silverstripe.org/en/contributing/coding_conventions/)
- [x] This change is covered with tests (or tests aren't necessary for this change)
- [x] Any relevant User Help/Developer documentation is updated; for impactful changes, information is added to the changelog for the intended release
- [x] CI is green

### CODE REVIEW

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
The comments in https://github.com/silverstripe/silverstripe-cms/pull/2993#pullrequestreview-2249599581 apply to this PR as well.

**Comment by satrun77** [comment]
I have updated the fix by moving the selection to the link element. This does not need the other PR. I have tested this with internal and external linking, it worked for me as expected.

File: client/src/legacy/TinyMCE_sslink.js [hunk 1]
```diff
@@ -35,6 +35,9 @@ const plugin = {
 
     // Callback for opening the edit link dialog form
     function openLinkDialog() {
+      // Find "a" node, issue https://github.com/silverstripe/silverstripe-cms/issues/2439
```
Comment (ChloeSartorelli) [inline client/src/legacy/TinyMCE_sslink.js]:
```suggestion
      // Find "a" node (we might have clicked on a child element e.g. "span")
```
There's no need to list the issue - git blame will inevitably lead us back to this PR and then to that issue if we really need that level of context on this.

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
Thanks, this works great locally. Just one tiny change to make.

File: client/src/legacy/TinyMCE_sslink.js (hunk 1 above)
Comment (satrun77) [reply client/src/legacy/TinyMCE_sslink.js]: Done updated

**Review by GuySartorelli — APPROVED** [review]
LGTM, works well locally. Thanks for this!

**Pull request merged**